### PR TITLE
Feature/rw update endpoints

### PR DESCRIPTION
**Pull request by rwilliams659**
# Som Timer Pull Request Form

## Description
The following endpoints were updated based on the back-end implementing users. These are now accessed via an endpoint that includes the individual user id:
- logging in a user
- getting a user's settings
- updating a user's settings
- posting a full user session 

Tests were also updated accordingly. 
## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other, Please explain:
`Your message here`
## Has This Been Tested?
- [ ] No
- [X] Yes
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration:
`npm test`
# Message for reviewer:
`Your message here`
# Checklist:
- [X] My code has no unused/commented out code
- [X] My code has no console logs
- [X] I have performed a self-review of my own code
- [X] I have commented on my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/Components/App/App.js [hunk 1]
```diff
@@ -25,8 +25,8 @@ const App = () => {
   
   return (
     <>
-      {!user.token && <Login />}
-      {user.token &&
+      {!user.email && <Login />}
```
Comment (sienna-kopf) [inline src/Components/App/App.js:28]: Good idea on using Email for user conditional checks here! Thanks for all your help with this Oauth stuff @rwilliams659

File: src/Components/Login/Login.js [hunk 2]
```diff
@@ -20,27 +23,46 @@ const Login = () => {
...
+          console.log(err.response.status)
+        }
+      })
+  }
   
   const onSignIn = googleUser => {
     const userName = googleUser.getBasicProfile().getName()
-    const token = googleUser.getAuthResponse().id_token
-    loginUser(userName, token)
+    const email = googleUser.getBasicProfile().getEmail()
```
Comment (sienna-kopf) [inline src/Components/Login/Login.js:47]: Glad we could retrieve the email so seamlessly from googleUser.

**Review by sienna-kopf — APPROVED** [review]
Thanks for getting this whipped up @rwilliams659 !! Looks great to me!

File: src/apiCalls.js [hunk 3]
```diff
@@ -1,36 +1,38 @@
 import axios from 'axios'
 const rootUrl = 'https://som-timer-be.herokuapp.com/api'
 
-export const getSettings = () => {
-  return axios.get(`${rootUrl}/timers/1`)
+export const getSettings = (userId) => {
```
Comment (jkwest-93) [inline src/apiCalls.js:4]: Nice work getting user specific network requests going!!

File: src/Components/App/App.test.js [hunk 4]
```diff
@@ -639,23 +625,23 @@ describe('App', () => {
...
 
     const { getByRole } = render(
       <MemoryRouter>
-        <UserContext.Provider
-          value={[{ userName: "Fran", token: "200", userId: 3 }]}
-        >
-          <SettingsProvider>
+        <SettingsContext.Provider value={[{ workInterval: "45", breakInterval: "5", sound: "reverbSplash", moodRating: true }, jest.fn()]}>
+          <UserContext.Provider value={[{ userName: "Fran", email: "fran@gmail.com", userId: 3 }]}
+          >
```
Comment (Abdeboskey) [inline src/Components/App/App.test.js:635]: Way to use this technique to trick the tests. Thank you for putting in so much time and energy to make this work!

File: src/Components/Login/Login.js [hunk 5]
```diff
@@ -20,27 +23,46 @@ const Login = () => {
...
+          const moodRating = response.data.mood
+          setSettings({ workInterval, breakInterval, sound, moodRating })
+        }
+      })
+      .catch(err => {
+        if (err.response) {
+          console.log(err.response.status)
+        }
+      })
+  }
```
Comment (Abdeboskey) [inline src/Components/Login/Login.js:43]: That looks awesome, nice work!

File: src/apiCalls.js [hunk 6]
```diff
@@ -1,36 +1,38 @@
...
+  return axios.get(`${rootUrl}/users/${userId}/timer`)
 }
 
-export const updateSettings = (settings) => {
-  return axios.put(`${rootUrl}/timers/1`, {
+export const updateSettings = (settings, userId) => {
+  return axios.put(`${rootUrl}/users/${userId}/timer`, {
```
Comment (Abdeboskey) [inline src/apiCalls.js:9]: Love me a good ol' dynamic endpoint like this!!

File: src/index.js [hunk 7]
```diff
@@ -11,15 +11,15 @@ import * as serviceWorker from './serviceWorker'
...
+      <UserProvider>
         <ViewProvider>
           <SessionProvider>
             <App />
           </SessionProvider>
         </ViewProvider>
-      </SettingsProvider>
-    </UserProvider>
+      </UserProvider>
+    </SettingsProvider>
```
Comment (Abdeboskey) [inline src/index.js:22]: I'm really happy that switching these around worked like this! Great job figuring this out 🤘🏼

**Review by Abdeboskey — APPROVED** [review]
Awesome work Rachel, this was much more complicated than just simply updating the endpoints, way to go figuring it out!

**Pull request merged**